node-js-fraud-example/gremlin.js

```diff
@@ -0,0 +1,292 @@
+            .property("accountId", `A${startingVertIndex + i + 1}`)
+            .property("balance", bal)
+            .property("bank", bank)
+            .next()
+        hubsArr.push(vert)
+    }
+
+    //Connect accounts and Shady people
+    for (let i = 0; i < numHubs; i++) {
+        await g
+            .addE("owns")
+            .from_(accountsArr[i].value)
+            .to(hubsArr[i].value)
+            .property("since", `${2020 + randomInt(-15, 5)}`)
+            .iterate()
+    }
+
+    //Make Transactions
+    for (let i = 0; i < numHubs; i++) {
+        let goonsAmt = randomInt(5, 8)
+        let goons = await g.V().hasLabel("Account").sample(goonsAmt).toList()
+        for (const goon of goons) {
+            for (let j = 0; j < randomInt(9, 25); j++) {
+                const amt = randomInt(5000, 30000);
+                const txId = `T${startingTransIndex + i + 1}`;
+                const type = Math.random() < 0.5 ? "debit" : "credit";
+                const year = String(randomInt(2000, 2025))
+                const month = String(randomInt(1, 12)).padStart(2, "0");
+                const day = String(randomInt(1, 28)).padStart(2, "0");
+                const date = `${year}-${month}-${day}`;
+                const device = devices[randomInt(0, devices.length - 1)];
+                await g
+                    .addE("Transaction")
+                    .from_(hubsArr[i].value)
+                    .to(goon)
+                    .property("transactionId", txId)
+                    .property("amount", amt)
+                    .property("device", device)
+                    .property("type", type)
+                    .property("timestamp", convertTimestampToLong(date))
+                    .iterate();
+            }
+        }
+    }
+
+}
+
+//Returns the given top amount of accounts in terms of outgoing
+// null amount gives whole list of vertices
+export async function rankMostTraffic(amount = null) {
+    const lists = await g.V()
+        .hasLabel("Account")
+        .project("accountId", "totalAmount", "ownerVert") //make map with keys
+        .by("accountId") //fill accountId bin with vertexes accountId prop
+        .by(__.coalesce( //run anonymous traversal and grab transactionEdges
+            __.outE("Transaction").values("amount").sum(), __.constant(0) //default to 0
+        ))
+        .by(__.in_("owns")
+            .hasLabel("User")
+            .limit(1)
+            .valueMap(true))
+        .toList();
+
+    lists.sort((a, b) => {
+        const totalA = a.get("totalAmount")
+        const totalB = b.get("totalAmount")
+        return totalB - totalA
+    })
+    if (amount) {
+        return lists.slice(0, amount)
+    }
+    return lists;
+}
+
+export async function getAllNames(name) {
+    if(name === ""){
+        return await g.V().hasLabel("User").values("name").toList()
+    }else{ // Grab all names of people affiliated with user
+        return await g.V().has("User", "name", name)
```

Comment:
you can leave it as is, but `bothE("Transaction").otherV()` is same as  `both("Transaction")`